cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/store/DatasetBasedTimeScheduleStore.java

```diff
@@ -89,11 +106,23 @@ public void initialize(ClassLoadHelper loadHelper, SchedulerSignaler schedSignal
       setMisfireThreshold(cConf.getLong(Constants.Scheduler.CFG_SCHEDULER_MISFIRE_THRESHOLD_MS));
       initializeScheduleTable();
       readSchedulesFromPersistentStore();
+      executorService = Executors.newSingleThreadExecutor(Threads.createDaemonThreadFactory(
+        "time-schedulestore-appver-upgrade-checker"));
+      executorService.submit(new UpgradeCompleteCheck(APP_VERSION_UPGRADE_KEY, factory, table, appVerUpgrade));
+      storeInitialized = true;
     } catch (Throwable th) {
       throw Throwables.propagate(th);
     }
   }
 
+  @Override
+  public void shutdown() {
```

Comment:
why is `DatasetBasedStreamSizeScheduleStore.shutdown()` synchronized but not necessary here?